### PR TITLE
[google-spreadsheet] Make WorksheetGridProperties partial

### LINKED ISSUE #48547: [google-spreadsheet] Properties in GridProperties should probably be marked optional
Shouldn't the properties in `interface WorksheetGridProperties` be marked optional? Right now TypeScript throws an error if all properties are not provided by default.

 - Authors: @EndBug, @the-vampiire

### PR DESCRIPTION
**Pull request by EndBug**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/EndBug/DefinitelyTyped.git>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Closes #48547

### CODE REVIEW

**Review by the-vampiire — COMMENTED** [review]
@EndBug i thought it would still be an interface but with `?:` on each field to mark optional. Is this syntax you've provided the more correct way? ill defer to your TS expertise here.

**Comment by EndBug** [comment]
@the-vampiire It's almost the same thing: the `Partial<T>` is like `T` but with every field as `prop?: originalType | undefined`
The difference is that this way you're allowed to explicitly set a property to `undefined`, which I think could be useful in some situations where you're merging different objects.

**Comment by the-vampiire** [comment]
very cool. thanks for the detail

**Review by the-vampiire — APPROVED** [review]

File: types/google-spreadsheet/index.d.ts
Comment (rbuckton) [inline types/google-spreadsheet/index.d.ts:75]: The problem with using `Partial<{ ... }>` instead of an interface is that error elaborations can become very messy:

image

vs.

image

I will approve and leave it up to you if it should merge as-is or if you want to change the definition.

**Review by rbuckton — APPROVED** [review]

File: types/google-spreadsheet/index.d.ts
Comment (the-vampiire) [reply types/google-spreadsheet/index.d.ts:75]: good catch there man. i see the value of the interface approach in this context. 

i'm not a TS dev so @EndBug  worked with me on making these. i'll defer the decision to him.

Comment (EndBug) [reply types/google-spreadsheet/index.d.ts:75]: My mistake, I'll switch back to an interface 👍

**Review by the-vampiire — APPROVED** [review]

**Comment by orta** [comment]
Looks reasonable to me  after reading https://github.com/DefinitelyTyped/DefinitelyTyped/issues/48547 👍🏻

**Pull request merged**
